backend/amd64/simd_selection.ml

```diff
@@ -73,23 +86,298 @@ let select_operation_sse op args =
     Some (Shuffle_32 i, args)
   | _ -> None
 
-let select_operation_sse2 _op _args = None
+let select_operation_sse2 op args =
+  match op with
+  | "caml_sse2_int8x16_add" -> Some (Add_i8, args)
+  | "caml_sse2_int16x8_add" -> Some (Add_i16, args)
+  | "caml_sse2_int32x4_add" -> Some (Add_i32, args)
+  | "caml_sse2_int64x2_add" -> Some (Add_i64, args)
+  | "caml_sse2_float64x2_add" -> Some (Add_f64, args)
+  | "caml_sse2_int8x16_add_saturating" -> Some (Adds_i8, args)
+  | "caml_sse2_int16x8_add_saturating" -> Some (Adds_i16, args)
+  | "caml_sse2_int8x16_add_saturating_unsigned" -> Some (Adds_u8, args)
+  | "caml_sse2_int16x8_add_saturating_unsigned" -> Some (Adds_u16, args)
+  | "caml_sse2_int8x16_sub" -> Some (Sub_i8, args)
+  | "caml_sse2_int16x8_sub" -> Some (Sub_i16, args)
+  | "caml_sse2_int32x4_sub" -> Some (Sub_i32, args)
+  | "caml_sse2_int64x2_sub" -> Some (Sub_i64, args)
+  | "caml_sse2_float64x2_sub" -> Some (Sub_f64, args)
+  | "caml_sse2_int8x16_sub_saturating" -> Some (Subs_i8, args)
+  | "caml_sse2_int16x8_sub_saturating" -> Some (Subs_i16, args)
+  | "caml_sse2_int8x16_sub_saturating_unsigned" -> Some (Subs_u8, args)
+  | "caml_sse2_int16x8_sub_saturating_unsigned" -> Some (Subs_u16, args)
+  | "caml_sse2_int8x16_max_unsigned" -> Some (Max_u8, args)
+  | "caml_sse2_int16x8_max" -> Some (Max_i16, args)
+  | "caml_sse2_float64x2_max" -> Some (Max_f64, args)
+  | "caml_sse2_int8x16_min_unsigned" -> Some (Min_u8, args)
+  | "caml_sse2_int16x8_min" -> Some (Min_i16, args)
+  | "caml_sse2_float64x2_min" -> Some (Min_f64, args)
+  | "caml_sse2_float64x2_mul" -> Some (Mul_f64, args)
+  | "caml_sse2_float64x2_div" -> Some (Div_f64, args)
+  | "caml_sse2_vec128_and" -> Some (And_bits, args)
+  | "caml_sse2_vec128_andnot" -> Some (Andnot_bits, args)
+  | "caml_sse2_vec128_or" -> Some (Or_bits, args)
+  | "caml_sse2_vec128_xor" -> Some (Xor_bits, args)
+  | "caml_sse2_vec128_movemask_8" -> Some (Movemask_8, args)
+  | "caml_sse2_vec128_movemask_64" -> Some (Movemask_64, args)
+  | "caml_sse2_vec128_shift_left_bytes" ->
+    let i, args =
+      extract_constant args 0 15 "caml_sse2_vec128_shift_left_bytes"
```

Comment:
```suggestion
      extract_constant args ~max:15 op
```
Same everywhere below, it seems `extract_constant` is always called with lower bound set to `0`.

Reply:
Updated